src/pytds/tds.py

```diff
@@ -1323,6 +1341,9 @@ def parse_prelogin(self, octets, login):
             elif type_id == PreLoginToken.INSTOPT:
                 # ignore instance name mismatch
                 pass
+            elif type_id == PreLoginToken.FEDAUTHREQUIRED:
```

Comment:
I suspect that with this change, even Sql User/Password auth will start erroring out, if the access_token is not set. 
You need to check if the payload has a 0 or a 1 value, which are considered valid. 
If the server sends a 0x01 AND if the Access_token is set, then only can we send the access token during login negotiation.